src/compositor/output/output.rs

```diff
@@ -32,37 +34,55 @@ impl OutputHandler for Output {
                     }
                 }
             });
+            let (lx, ly) = with_handles!([(cursor: {&cursor})] => {
+                cursor.coords()
+            }).unwrap();
+            for drag_icon in &seat.drag_icons {
+                with_handles!([(drag_icon: {&drag_icon.handle})] => {
+                    let (sx, sy) = drag_icon.position();
+                    render_surface(&mut renderer, layout, &mut drag_icon.surface(), lx as i32 + sx, ly as i32 + sy);
+                    }).unwrap();
+            }
         }).unwrap();
     }
 }
 
+fn render_surface(renderer: &mut Renderer,
+                  layout: &mut OutputLayoutHandle,
+                  surface: &mut SurfaceHandle,
+                  lx: i32,
+                  ly: i32) {
+    with_handles!([(surface: {surface}), (layout: {&mut *layout})] => {
+        let (width, height) = surface.current_state().size();
+        let (render_width, render_height) =
+            (width * renderer.output.scale() as i32,
+            height * renderer.output.scale() as i32);
+        let render_box = Area::new(Origin::new(lx, ly),
+        Size::new(render_width,
+                  render_height));
```

Comment:
Unnecessary newline

Reply:
Wait...no bad indentation? Didn't realize that was part of `Area::new` call until I saw that extra `)`.